src/RouteGenerator.js

```diff
@@ -0,0 +1,18 @@
+import * as Immutable from 'immutable';
+
+export default class RouteGenerator {
+  constructor() {
+    this._routeToIndexMap = new WeakMap();
+  }
+
+  getRouteFor(branch) {
+    return '$$_' + branch.map(route => {
+      let id = this._routeToIndexMap.get(route);
+      if (!id) {
+        id = this._routeToIndexMap.length;
```

Comment:
I don't think this is quite right. MDN says `WeakMap#length` is always 0: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap#Properties

Maybe the right way to do this would be to have a separate counter for the next free index, to avoid any possible collisions. I think you also want to add `"optional": ["runtime"]` in `.babelrc` to avoid breaking IE < 11 which don't have native `WeakMap` support.